test/integration/restart_task_test.rb

```diff
@@ -117,8 +162,45 @@ def test_restart_not_existing_namespace
       namespace: "walrus",
       logger: logger
     )
-    refute restart.perform(["web"])
-    assert_logs_match("Namespace `walrus` not found in context `minikube`. Aborting the task.")
+    assert_restart_failure(restart.perform(["web"]))
+    assert_logs_match_all([
+      "Result: FAILURE",
+      "Namespace `walrus` not found in context `minikube`"
+    ],
+      in_order: true)
+  end
+
+  def test_restart_failure
+    success = deploy_fixtures("hello-cloud", subset: ["configmap-data.yml", "web.yml.erb"]) do |fixtures|
+      deployment = fixtures["web.yml.erb"]["Deployment"].first
+      deployment["spec"]["progressDeadlineSeconds"] = 12
+      container = deployment["spec"]["template"]["spec"]["containers"].first
+      container["readinessProbe"] = {
```

Comment:
Alternative ideas for simulating a failed restart greatly appreciated!

Reply:
This seems reasonable to me.